packages/task/src/common/task-watcher.ts

```diff
@@ -42,6 +43,9 @@ export class TaskWatcher {
             },
             onDidProcessTaskOutput(event: TaskOutputProcessedEvent): void {
                 outputProcessedEmitter.fire(event);
+            },
+            onBackgroundTaskActive(event: BackgroundTaskEndEvent): void {
```

Comment:
What active means? 

coding guidelines for event naming: https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#event_names

Reply:
was fixed and was change to 'onBackgroundTaskEnded'.
The server fires  'BackgroundTaskEndedEvent' when a background task matches its 'beginsPattern' and 'endsPattern' properties in order to notify the client (that the backround task is active).
e.g. if we have a main task that in its "dependsOn" array there is a task array that for example runs tomcat ( so it is running in the background and doesn't seems to have exit code ) , then the root task will never start.
So the solution is to have a problem matcher with a background property and when it matches the 'beginsPattern' and 'endsPattern' properties , we know that the task is active and we can move forward to the main task.